lib/matplotlib/contour.py

```diff
@@ -1213,10 +1211,17 @@ def _process_levels(self):
         # (Colorbar needs this even for line contours.)
         self._levels = list(self.levels)
 
+        tempLocator = Locator()
+        if self.logscale:
+            tempLocator = LogLocator()
+        if self.locator is not None:
+            tempLocator = self.locator
```

Comment:
Instead of using a temporary Locator and then requiring that all Locators have 3 new private methods, I think it would be better to keep everything within the ContourBase class.  This will be more readable than using Locator private methods that modify one of their arguments.  It will likely require fewer LOC as well.